tests/keithley_2700.py

```diff
@@ -150,13 +153,241 @@ def test_WHEN_end_channel_range_set_THEN_end_channel_matches_the_set_state(self)
         for channel in sample_channels:
             self.ca.assert_setting_setpoint_sets_readback(channel, "CH:END")
 
-    def test_WHEN_measurement_mode_set_THEN_mesaurement_mode_matches_the_set_state(self):
+    def test_WHEN_measurement_mode_set_THEN_measurement_mode_matches_the_set_state(self):
         self.ca.set_pv_value("CH:START:SP", 101)
         self.ca.set_pv_value("CH:END:SP", 210)
         sample_data = {0: "VOLT:DC", 1: "VOLT:AC", 2: "CURR:DC", 3: "CURR:AC", 4: "RES", 5: "FRES", 6: "CONT", 7: "FREQ", 8: "PER"}
         for measurement_enum, measurement_string in sample_data.items():
-            self.ca.assert_setting_setpoint_sets_readback(measurement_string, "MEASUREMENT", expected_value=measurement_string)
+            self.ca.assert_setting_setpoint_sets_readback(measurement_string, "MEASUREMENT",
+                                                          expected_value=measurement_string)
 
     def test_WHEN_elements_set_THEN_elements_are_as_expected(self):
         elements = "READ, CHAN, TST"
         self.ca.assert_setting_setpoint_sets_readback(elements, "DATAELEMENTS")
+
+
+class BufferTests(unittest.TestCase):
+    def setUp(self):
+        self._lewis, self._ioc = get_running_lewis_and_ioc("keithley_2700", DEVICE_PREFIX)
+        self.ca = ChannelAccess(default_timeout=30, device_prefix=DEVICE_PREFIX)
+        self.ca.assert_that_pv_exists("IDN")
+        self.ca.set_pv_value("BUFF:CLEAR:SP", "")
+        self.ca.assert_that_pv_is("BUFF:AUTOCLEAR", "ON")
+
+    def _generate_readings(self, num_readings_gen, time_between):
+        min_r = 1000
+        max_r = 2000
+        start_time = 1
+        channels = [c for c in range(101, 110 + 1)] + [c for c in range(201, 210 + 1)]
+        readings = []
+
+        for i in range(num_readings_gen):
+            resistance = min_r + i
+            while resistance > max_r:
+                resistance = resistance - min_r
+            time_stamp = start_time + (i*time_between)
+            channel = channels[i % 20]
+            readings.append(str(resistance) + "," + str(time_stamp) + "," + str(channel))
+
+        return readings
+
+    def _set_buffer_size(self, buff_size):
+        self.ca.set_pv_value("BUFF:SIZE:SP", buff_size)
+        self.ca.assert_that_pv_is("BUFF:SIZE", buff_size)
+
+    @contextmanager
```

Comment:
Context manager might be overkill because it does nothing on the `finally`, it can just be replaced by a function - Oram, Dom (2019)